ocaml/libs/tgroup/tgroup.mli

```diff
@@ -0,0 +1,82 @@
+(*
+ * Copyright (C) Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+(** [Group] module helps with the classification of different xapi execution
+    threads.*)
+module Group : sig
+  (** Abstract type that represents a group of execution threads in xapi. Each
+  group corresponds to a Creator, and has a designated level of priority.*)
+  type t
+
+  (** Generic representation of different xapi threads originators. *)
+  module Originator : sig
+    (** Type that represents different originators of xapi threads. *)
+    type t = Internal_Host_SM | EXTERNAL | Internal_Server
+
+    val of_string : string -> t
```

Comment:
Whats strings are expected here?

Reply:
My intention is to have a new http header, `originator`, based on which I set the priority of requests. Initially `SM` would send this header and (the `CLI` would be added next). So this function would be called on the values of that specific header.

Not having a mapping to `Internal_server` was intentional, as I didn't want the possibility of an http request to mimic a Server priority.  Classifying the server threads would use `Originator.Internal_Server` directly.